### ISSUE TITLE
Backslash

### ISSUE THREAD

**Issue by polilla** [tablacus/TablacusExplorer#541]
It's no more posible to enter the backslash symbol "\" into the address bar on my german keyboard.
Latest version of TablacusExplorer already installed.

**Comment by polilla** [comment]
UNC path to my server is \\<computername>
It works when i use a slash instead of backslash.
But the correct UNC path in Windows is using backslash.
The key add-on not exists.

TE64 22.1.20 Microsoft Windows 11 Pro 64-Bit 21H2 (10.0.22000) IE/11 JS/11.0.16384 de_de 96 AMD Ryzen 7 3700X 8-Core Processor              Windows Defender

mouse 1.33,key 1.26,addressbar 1.70,filterbar 1.38,back 1.18,forward 1.18,up 1.20,treeview 1.43,toolbar 1.43,mainmenu 1.16,tabplus 1.91,multithread 1.28,remember 1.42,takeoverfoldersettings 1.08,addonsupdater 1.10,notepadalternative 1.03,segoeicons 1.03

![te](https://user-images.githubusercontent.com/11969556/150640394-0df66cd2-67ae-4a63-aeac-0c9106fde192.png)

**Comment by tablacus** [comment]
May work with Tablacus Explorer WebView2
https://tablacus.github.io/tewv2.html

Add-ons of WebView2
![image](https://user-images.githubusercontent.com/5156977/150641945-edec6676-6f65-4bbf-b58b-2169430f4ba3.png)

For Windows 10, WebView2 Runtime is required.

Regards,

**Comment by polilla** [comment]
No more problems with WebView2. Thank you.

**Comment by stephannn** [comment]
The error seems to be a change in the te64.dll, but no clue which cpp is doing the keyboard mapping https://github.com/tablacus/TablacusExplorer/tree/master/TE

**Comment by AlfBekov** [comment]
> May work with Tablacus Explorer WebView2 https://tablacus.github.io/tewv2.html
> 
> Add-ons of WebView2 ![image](https://user-images.githubusercontent.com/5156977/150641945-edec6676-6f65-4bbf-b58b-2169430f4ba3.png)
> 
> For Windows 10, WebView2 Runtime is required.
> 
> Regards,

It's happenning also with a spanish keyboard, but installing WebView2 isn't a valid solution on portable environments.

Seems like Alt and AltGr keys no longer work (as Alt+92 could've been a temporary workaround) anywhere typing is allowed (i.e. the search complements bar in Tools->Complements, the Everything search bar if this complement is installed and even in the language box in Tools->Options->General).

Disabling all complements doesn't solve the issue. 

I've checked that everything works ok up until version 22.1.7, and stops doing it since 22.1.10 up to current version (24.3.28) in either 32 and 64 bits. I'd blame #532, but I have no proof... either way I can't reproduce the "ding" issue in 22.1.7 nor 24.3.28.

**Comment by lrq3000** [comment]
Same problem here, AltGr does not work. The problem is that I am using a Bepo keyboard mapping, so that AltGr is necessary to input underscores ("_") characters. Hence, in TablacusExplorer, all Bepo keyboard users cannot input underscores. This is literally the only thing that prevents me from completely switching to Tablacus as my default explorer.

I am not using the WebView2 version, just the win64 version that is offered to install as a portable zip. I am also using the latest version 24.12.9

**Comment by ccfs** [comment]
I think that the cause is at function GetKeyShift of sync.js. I think that the function indifferentiates left and right Shift/Ctrl/Alt/Win keys. However on some Western language keyboards, the right Alt key is actually AltGr key, which should be differentiated from the left Alt key in order to correctly produce a character.

A possible solution is to add an "AltGr key" option to TE (under Options > General > Language) and amend GetKeyShift function accordingly. However the programmer should be using such a keyboard and language edition of Windows; otherwise there is no way to test. And I cannot help further.

**Comment by lrq3000** [comment]
@ccfs Thank you very much for your insights. I am an experienced programmer but not of the stack used for this app, but maybe I can do something with the help of genAI. If this issue continues to itch me, I will try to compile and if this works I'll try to fix the issue.

**Comment by tablacus** [comment]
Please try updating to Tablacus Explorer v24.12.23.

**Comment by ccfs** [comment]
[lrq3000](https://github.com/lrq3000), does the new version solve the problem?

**Comment by lrq3000** [comment]
@tablacus @ccfs Thank you so much guys for your responsiveness on this issue! Unfortunately it is not fixd in v24.12.23. But I did some more troubleshooting.

The AltGr issue happens only in the rest of the interface but not the main files explorer. Ie, I can use AltGr when renaming files, but it does not work in the address bar nor in the filter bar.

I tried to add the WebView2 lib (I did not know it was just an extension of the main app), and it does work then! So the WebView2 lib does fix the issue.

I am totally ok to use the WebView2 version, so for me this issue is fine being fixed this way, but I can test further if you would like to try other changes.

But I would like to suggest, why not make the WebView2 version the default one, why is the WebView2 lib not bundled in the main release since it seems to offer more features/being more stable?

**Comment by ccfs** [comment]
1. Microsoft WebView2 runtime, which is required by TE WebView2, is not installed by default on older versions of Windows.
2. WebView2 version has a lot of problems. You will find them out when you use it.

**Comment by lrq3000** [comment]
Arg, so no perfect solution yet for this solution, that's a bummer.

Well, if you guys want to try other fixes, I will happily try them. I'm unfortunately not experienced enough with the codebase or C++ to debug this effectively...

**Comment by garry-ut99** [comment]
> lrq3000: Well, if you guys want to try other fixes, I will happily try them. 

Me too as well is ready to help and test.

**Comment by garry-ut99** [comment]
> stephannn: The error seems to be a change in the te64.dll, but no clue which cpp is doing the keyboard mapping https://github.com/tablacus/TablacusExplorer/tree/master/TE

> AlfBekov: I've checked that everything works ok up until version 22.1.7, and stops doing it since 22.1.10 up to current version (24.3.28) in either 32 and 64 bits.



I agree, I've found out that in v22.2.10 replacing `te64.dll` with the one from 22.1.7, fixes the diacritics issue, therefore is seems to be [THIS](https://github.com/tablacus/TablacusExplorer/commit/4a1fced5cb5d4724e15f52241ac53ff3f0bca7bc?diff=unified#diff-61b636c7c5250502f07af8e097e87fe62a3156b20b60c3d35e623874d7e0b0a7L548-R548) change in `TE.cpp` to be responsible for the issue. 

There is no difference in size between `te64.dll` in 22.1.7 and 22.1.10 (0 bytes) - but the hashes are different, which indicates that there were some subtle in-place replacements / changes introduced in `te64.dll`, either by the changed code line or by the compiler optimizations. WinMerge binary comparison shows a few subtle changes is several places. 

Someone says that the changed code line in `TE.cpp` is faulty and that there was a bug introduced:
> The change from `if (pActiveObject->TranslateAcceleratorW(pMsg) == S_OK)` to `if SUCCEEDED(...)` in `TE.cpp` is causing issues with diacritics (AltGr combinations). `TranslateAcceleratorW` returns `S_FALSE` when a key is NOT an accelerator, and it should be processed further. `SUCCEEDED` catches both `S_OK` and `S_FALSE`, effectively "swallowing" these key strokes.

**Comment by polilla** [comment]
Thank you that you have solved the AltGr issue with 26.4.4 after that long time.
But it not works for 100 %.
When i connect to my other computer with VNC (in my case TightVNC) the AltGr works fine in Windows file explorer but still not yet in TablacusExplorer.

**Comment by garry-ut99** [comment]
You'd be better off stopping the trolling/manipulation and sticking to facts instead of fiction, and thanking all people who deserve it: 
- it was the developer who broke this feature in the first place, and didn’t know how to fix it since then
- I was the one who solved the problem (using information provided by stephannn and AlfBekov), and the developer simply implemented my solution.

> polilla: But it not works for 100 %. (...)

[26.4.11](https://github.com/tablacus/TablacusExplorer/releases/tag/26.4.11) is out, perhaps it's completely fixed now.

**Comment by polilla** [comment]
Version 26.4.11 now works fine here with TightVNC. No more problems.
Thanks to ALL for that solution.

**Comment by garry-ut99** [comment]
OK, although some users (including me) who use [the Patched version 25.6.8](https://github.com/tablacus/TablacusExplorer/issues/692), will still need to wait for the Patch release supporting v26.4.11, because the current version of the Patch says it works only with: "_TE version must be 25.6.8._", and there is no "_or higher_" (unless it was forgotten to write "_or higher_".)

**Comment by garry-ut99** [comment]
The thread can be closed now.
